src/components/ViewRequest/Documents/DocumentView.tsx

```diff
@@ -0,0 +1,125 @@
+import { Icon } from "@fluentui/react";
+import {
+  Button,
+  Caption1,
+  Card,
+  CardHeader,
+  Link,
+  Spinner,
+  Title3,
+} from "@fluentui/react-components";
+import { getFileTypeIconProps } from "@fluentui/react-file-type-icons";
+import { DeleteIcon } from "@fluentui/react-icons-mdl2";
+import { SPDocument, useDeleteDocument } from "api/documentsApi";
+
+export const DocumentView = (props: {
+  document: SPDocument;
+  readonly?: boolean;
+}) => {
+  const deleteDocument = useDeleteDocument(props.document);
+
+  const extension = props.document.ServerRelativeUrl.substring(
+    props.document.ServerRelativeUrl.lastIndexOf(".") + 1
+  );
+
+  const lastModified = new Date(
+    props.document.TimeLastModified
+  ).toLocaleString();
+
+  const isOfficeFile: boolean = wordExtensions
+    .concat(excelExtensions)
+    .concat(ppExtensions)
+    .includes(extension);
+
+  const viewEdit = props.readonly ? "v" : "e";
+
+  //Application specific URI schemes
+  let downloadUrl = props.document.ServerRelativeUrl;
+  if (wordExtensions.includes(extension)) {
+    downloadUrl = `ms-word:of${viewEdit}|u|${window.location.origin}${props.document.ServerRelativeUrl}`;
+  } else if (excelExtensions.includes(extension)) {
+    downloadUrl = `ms-excel:of${viewEdit}|u|${window.location.origin}${props.document.ServerRelativeUrl}`;
+  } else if (ppExtensions.includes(extension)) {
+    downloadUrl = `ms-powerpoint:of${viewEdit}|u|${window.location.origin}${props.document.ServerRelativeUrl}`;
+  }
+
+  return (
+    <Card style={{ margin: "0.5em" }}>
+      <CardHeader
+        image={
+          <Icon
+            {...getFileTypeIconProps({
+              extension: extension,
+              size: 64,
+              imageFileType: "png",
+            })}
+          />
+        }
+        header={
+          <Link download={!isOfficeFile} href={encodeURI(downloadUrl)}>
+            <Title3 className="document-name">{props.document.Name}</Title3>
+          </Link>
+        }
+        description={
+          <Caption1>
+            Last Updated By: {props.document.ModifiedBy.Title}
+            <br />
+            Last Updated On: {lastModified}
+          </Caption1>
+        }
+        action={
+          <>
+            {!deleteDocument.isLoading && (
+              <Button
```

Comment:
Add a confirmation box to ensure they want to delete.  isSuccess and isError handling thoughts?